src/box/lua/experimental/connpool.lua

```diff
@@ -0,0 +1,64 @@
+local config = require('config')
+local checks = require('checks')
+local netbox = require('net.box')
+
+local connections = {}
+
+local function is_connection_valid(conn, opts)
+    if conn == nil or conn.state == 'error' or conn.state == 'closed' then
+        return false
+    end
+    assert(type(opts) == 'table')
+    local conn_opts = conn.opts or {}
+    if opts.fetch_schema == true and conn_opts.fetch_schema == false then
+        return false
+    end
+    return true
+end
+
+local function connect_using_config(instance_name, opts)
+    checks('string', {
+        connect_timeout = '?number',
+        wait_connected = '?boolean',
+        fetch_schema = '?boolean',
+    })
+    opts = opts or {}
+
+    local conn = connections[instance_name]
+    if not is_connection_valid(conn, opts) then
+        local uri_opts = {
+            instance = instance_name,
+            show_password = true,
```

Comment:
AFAIR, this `:instance_uri()` option is gone (was present in a draft patch version).

Reply:
True, fixed.